sled-agent/src/bootstrap/trust_quorum/share_distribution.rs

```diff
@@ -0,0 +1,78 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde::{Deserialize, Serialize};
+use serde_json;
+use std::fs;
+use std::path::{Path, PathBuf};
+use vsss_rs::Share;
+
+use super::rack_secret::Verifier;
+use crate::bootstrap::agent::BootstrapError;
+
+const FILENAME: &'static str = "share.json";
+
+/// A ShareDistribution is an individual share of a secret along with all the
+/// metadata required to allow a server in possession of the share to know how
+/// to correctly recreate a split secret.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct ShareDistribution {
+    pub threshold: usize,
+    pub total_shares: usize,
+    pub verifier: Verifier,
+    pub share: Share,
+}
+
+impl ShareDistribution {
+    pub fn write(&self, dir: &Path) -> Result<(), BootstrapError> {
```

Comment:
This is a nitpick, but as a heads up for both this read/write method...

If you use the following:
```suggestion
    pub fn write<P: AsRef<Path>>(&self, dir: P) -> Result<(), BootstrapError> {
        let dir = dir.as_ref();
```

These functions will be callable with a much more flexible set of types. This is a pattern followed pretty extensively by the standard library (example: https://doc.rust-lang.org/1.29.1/std/fs/struct.File.html#method.open).

I dunno how many callers we're gonna have, but it seems kinda handy for making this API a little more usable.

Reply:
Good idea! Done.